statsmodels/formula/_manager.py

```diff
@@ -0,0 +1,452 @@
+
+        Returns
+        -------
+        {EvalEnvironment, dict}
+            A formula-engine-dependent empty evaluation environment.
+        """
+        if self._engine == "patsy":
+            from patsy.eval import EvalEnvironment
+
+            return EvalEnvironment({})
+        else:
+            return {}
+
+    @property
+    def intercept_term(self):
+        """
+        Get the formula-engine-specific intercept term.
+
+        Returns
+        -------
+        Term
+            The intercept term.
+        """
+        if self._engine == "patsy":
+            from patsy.desc import INTERCEPT
+
+            return INTERCEPT
+        else:
+            from formulaic.parser.types.factor import Factor
+            from formulaic.parser.types.term import Term
+
+            return Term((Factor("1"),))
+
+    def remove_intercept(self, terms):
+        """
+        Remove intercept from Patsy terms.
+
+        Parameters
+        ----------
+        terms : list[Term]
+            A list of terms that might have an intercept
+
+        Returns
+        -------
+        list[Term]
+            The terms with the intercept removed, if present.
+        """
+        intercept = self.intercept_term
+        if intercept in terms:
+            terms.remove(intercept)
+        return terms
+
+    def has_intercept(self, spec):
+        """
+        Check if the model specification has an intercept term.
+
+        Parameters
+        ----------
+        spec
+
+        Returns
+        -------
+        bool
+            True if the model specification has an intercept term, False otherwise.
+        """
+
+        return self.intercept_term in spec.terms
+
+    def intercept_idx(self, spec):
+        """
+        Returns boolean array index indicating which column holds the intercept.
+
+        Parameters
+        ----------
+        spec
+
+        Returns
+        -------
+        ndarray
+            Boolean array index indicating which column holds the intercept.
+        """
+        from numpy import array
+
+        intercept = self.intercept_term
+        return array([intercept == i for i in spec.terms])
+
+    def get_na_action(self, action: str = "drop", types: Sequence[Any] = _NoDefault):
+        """
+        Get the NA action for the formula engine.
+
+        Parameters
+        ----------
+        action
+        types
+
+        Returns
+        -------
+        NAAction | str
+            The formula-engine-specific NA action.
+
+        Notes
+        -----
+        types is ignored when using formulaic.
+        """
+        types = ["None", "NaN"] if types is _NoDefault else types
+        if self._engine == "patsy":
+            return NAAction(on_NA=action, NA_types=types)
+        else:
+            return action
+
+    def get_spec(self, formula):
+        if self._engine == "patsy":
+            return patsy.ModelDesc.from_formula(formula)
+        else:
+            return formulaic.Formula(formula)
```

Comment:
if you want to be able to set other parts of the spec, you can also do: `ModelSpec(formula=formula)`.

Reply:
Ah... whoops. For nested formulae you'd want: `ModelSpec.from_spec(formula)`; which then returns a `ModelSpecs` instance as necessary.